packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx

```diff
@@ -39,41 +43,64 @@ export class AtomicSearchInterface {
   @Prop() sample = false;
   @Prop({reflect: true}) pipeline = 'default';
   @Prop({reflect: true}) searchHub = 'default';
-  @Prop() logLevel?: LogLevel = 'silent';
+  @Prop() logLevel?: LogLevel;
   @Prop() i18n: i18n = i18next.createInstance();
+  @Prop({reflect: true}) language = 'en'; // TODO: make watchable and update i18next language on change
   @Prop({mutable: true}) engine?: Engine;
   @State() error?: Error;
+  @Event() ready!: EventEmitter;
 
   private unsubscribe: Unsubscribe = () => {};
   private hangingComponentsInitialization: InitializeEvent[] = [];
-  private initialized = false;
-  private afterInitializationCallbacks: (() => void)[] = [];
+  private systemsToInitialize = {
+    i18next: false,
+    headless: false,
+  };
 
   componentWillLoad() {
+    this.i18n
```

Comment:
Small nitpick, but I would put this in a `initializei18()` method.

Just so that we get a nice: 

```
componentWillLoad() {
  this.initializei18n();
  this.initializeHeadless();
  this.initializeSystemFoo();
  etc.
}

```

Reply:
Also, I'm not sure why there's a mix of `componentWillLoad()`, and then we also have an `initialize` method.

Can't it just be inside a single `initialize()` method that does everything ? And each system has safeguards in place to make sure they are in a correct state to be initialized when the method is called/wait until everything is in place before resolving.

Otherwise, at first glance it's a bit spaghetti ? As in there's multiple "entry point" that all do some part of the initialization job.

Ideally there should be one entry point that initialize every system one by one, in a declarative manner.